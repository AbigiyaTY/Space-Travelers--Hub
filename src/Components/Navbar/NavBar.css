.nav-menu {
  display: flex;
  justify-content: space-evenly;
}

.header {
  display: flex;
  border-bottom: solid rgb(202, 198, 198);
  border-width: 2px;
}

h1 {
  align-self: center;
}

.nav-bar {
  align-self: center;
  display: flex;
  justify-content: space-between;
  width: 100%;
}

.nav-menu li {
  align-self: center;
  list-style-type: none;
  margin: 0;
  text-decoration: none;
  padding: 0 2vw 0 2vw;
}

.logo {
  width: 100px;
}

.nav-menu li:nth-child(2) {
  border-right: 2px solid black;
}

.rockets,
.mission,
.my-profile {
  text-decoration: none;
}

.active {
  /* color: rgb(218, 42, 209); */
  text-decoration: underline;
}
