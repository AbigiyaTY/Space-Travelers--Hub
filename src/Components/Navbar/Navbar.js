import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './Navbar.module.css';
// import the nav image
import NavLogo from './planet.png';

const Navbar = () => (
  <nav className={styles['nav-container']}>
    <div className={styles['nav-logo']}>
      <img className={styles['logo-img']} src={NavLogo} alt="planet" />
      <h1 className={styles['logo-title']}>`Space Travelers' Hub`</h1>
    </div>
    <ul className={styles['nav-list']}>
      <li className={styles['list-item']}>
        <NavLink className={styles['nav-link']} to="/" exact>
          Rockets
        </NavLink>
      </li>
      <li className={styles['list-item']}>
        <NavLink className={styles['nav-link']} to="/missions">
          Missions
        </NavLink>
      </li>
      <span> | </span>
      <li className={styles['list-item']}>
        <NavLink className={styles['nav-link']} to="/profile">
          My Profile
        </NavLink>
      </li>
    </ul>
  </nav>
);

export default Navbar;
