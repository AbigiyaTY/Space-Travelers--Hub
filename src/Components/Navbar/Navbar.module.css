.nav-container {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background-color: #fff;
  text-align: center;
  border-bottom: solid 1px #e8e8e8;
  height: 4rem;
  width: 100%;
  padding: 0 2rem;
  margin: 0;
}

.nav-logo {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    height: 100%;
    width: 100%;
    padding: 0 2rem;
    margin: 0;
}

.logo-img {
    height: 100%;
    width: 100px;
    padding: 0 2rem;
    margin: 0;
}

.logo-title {
  display: flex;
  /* font-family: var(--montserrat); */
  font-size: 1.875rem;
  font-weight: 800;
  font-weight: bold;
  color: #0290ff;
  text-align: justify;
  min-width: 16rem;
  margin: 0;
  padding: 0;
}

.nav-list {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  text-align: center;
  gap: 1rem;
  list-style: none;
  width: 100%;
  padding: 0;
  margin: 0;
}

.list-item {
  display: flex;
  /* font-family: var(--montserrat); */
  font-size: 0.9rem;
  letter-spacing: 1.9px;
  text-decoration: none;
  color: #121212;
  padding: 0.5rem;
}

.nav-link {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    text-align: center;
    text-decoration: none;
    color: #121212;
    padding: 0.5rem;
}

.nav-link:active {
    text-decoration: underline;
}
