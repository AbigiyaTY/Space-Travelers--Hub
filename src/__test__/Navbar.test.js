import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../Components/Navbar/Navbar';

it('renders correctly', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
