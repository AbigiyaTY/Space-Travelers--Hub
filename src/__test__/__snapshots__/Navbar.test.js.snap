// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="header p-2"
>
  <img
    alt="logo"
    className="logo"
    src="logo.png"
  />
  <nav
    className="nav-bar"
  >
    <h1>
      Space Travelers' Hub
    </h1>
    <ul
      className="nav-menu"
    >
      <li>
        <a
          activeclassname="active"
          className="rockets"
          href="/rockets"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          activeclassname="active"
          className="mission "
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li>
        <a
          activeclassname="active"
          className="my-profile"
          href="/myprofile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
